fv3gfs/physics/stencils/fv_update_phys.py

```diff
@@ -89,27 +89,32 @@ class ApplyPhysics2Dycore:
     """
 
     def __init__(
-        self, grid, namelist, comm: fv3gfs.util.CubedSphereCommunicator, grid_info,
+        self,
+        stencil_factory: StencilFactory,
+        grid,
+        namelist,
+        comm: fv3gfs.util.CubedSphereCommunicator,
+        grid_info,
     ):
         self.grid = grid
         self.namelist = namelist
         self.comm = comm
         self._dt = Float(self.namelist.dt_atmos)
-        self._moist_cv = FrozenStencil(
+        self._moist_cv = stencil_factory.from_origin_domain(
             moist_cv,
             origin=self.grid.compute_origin(),
             domain=self.grid.grid_indexing.domain_compute(add=(0, 0, 1)),
```

Comment:
Doesn't need to be done in this PR if it needs to be expedited, but these simpler `from_origin_domain` calls could be refactored to use `from_dims_halo`, to remove dependence on `Grid`/`GridIndexing`.
```suggestion
        self._moist_cv = stencil_factory.from_dims_halo(
            moist_cv,
            dims=[fv3util.X_DIM, fv3util.Y_DIM, fv3util.Z_INTERFACE_DIM]
```

Where you do still use GridIndexing, you will eventually need to use `stencil_factory.grid_indexing` and not `grid.grid_indexing`.